src/ripple/app/ledger/impl/LedgerToJson.cpp

```diff
@@ -118,24 +121,68 @@ fillJsonTx(
     if (bBinary)
     {
         txJson[jss::tx_blob] = serializeHex(*txn);
+        if (fill.context->apiVersion > 1)
+            txJson[jss::hash] = to_string(txn->getTransactionID());
+
+        auto const json_meta =
+            (fill.context->apiVersion > 1 ? jss::meta_blob : jss::meta);
         if (stMeta)
-            txJson[jss::meta] = serializeHex(*stMeta);
+            txJson[json_meta] = serializeHex(*stMeta);
     }
     else
     {
-        copyFrom(txJson, txn->getJson(JsonOptions::none));
-        RPC::insertDeliverMax(txJson, txnType, fill.context->apiVersion);
-        if (stMeta)
+        if (fill.context->apiVersion > 1)
         {
-            txJson[jss::metaData] = stMeta->getJson(JsonOptions::none);
-
-            // If applicable, insert delivered amount
-            if (txnType == ttPAYMENT || txnType == ttCHECK_CASH)
-                RPC::insertDeliveredAmount(
-                    txJson[jss::metaData],
-                    fill.ledger,
-                    txn,
-                    {txn->getTransactionID(), fill.ledger.seq(), *stMeta});
+            std::string hash;
+            copyFrom(
+                txJson[jss::tx_json],
+                txn->getJson(
+                    JsonOptions::none, false, {std::optional(std::ref(hash))}));
+            txJson[jss::hash] = hash;
+            RPC::insertDeliverMax(
+                txJson[jss::tx_json], txnType, fill.context->apiVersion);
+
+            if (stMeta)
+            {
+                txJson[jss::meta] = stMeta->getJson(JsonOptions::none);
+
+                // If applicable, insert delivered amount
+                if (txnType == ttPAYMENT || txnType == ttCHECK_CASH)
+                    RPC::insertDeliveredAmount(
+                        txJson[jss::meta],
+                        fill.ledger,
+                        txn,
+                        {txn->getTransactionID(), fill.ledger.seq(), *stMeta});
+            }
+
+            const bool validated =
+                fill.context->ledgerMaster.isValidated(fill.ledger);
+            txJson[jss::validated] = validated;
+            if (validated)
+            {
+                txJson[jss::ledger_index] = to_string(fill.ledger.seq());
+                txJson[jss::ledger_hash] = to_string(fill.ledger.info().hash);
```

Comment:
As per [Recommendations](https://github.com/XRPLF/rippled/issues/4727),  `ledger_hash` is expected to be included in closed ledgers. The code includes this field in validated ledgers, not [closed ledgers](https://xrpl.org/open-closed-validated-ledgers.html).

Should we update the recommendations to reflect the code?
<img width="1067" alt="image" src="https://github.com/XRPLF/rippled/assets/21219765/1bf3082b-90a4-42c7-b0ef-709e11e6960b">

Reply:
Thanks, I implemented your suggestion.

Reply:
thank you for looking into it 👍